### PR TITLE
API Assignment-2

### PR DESCRIPTION
**Pull request by waghmaresuyog**

### CODE REVIEW

File: API_Auto/pom.xml [hunk 1]
```diff
@@ -3,15 +3,58 @@
...
+            <groupId>org.apache.logging.log4j</groupId>
+            <artifactId>log4j-core</artifactId>
+            <version>2.20.0</version>
+        </dependency>
+        <dependency>
+            <groupId>log4j</groupId>
+            <artifactId>log4j</artifactId>
+            <version>1.2.12</version>
+            <scope>test</scope>
+        </dependency>
```
Comment (anushreeSonak) [inline API_Auto/pom.xml]: remove extra space

File: API_Auto/pom.xml [hunk 2]
```diff
@@ -3,15 +3,58 @@
...
+        <!-- https://mvnrepository.com/artifact/org.testng/testng -->
+        <dependency>
+            <groupId>org.testng</groupId>
+            <artifactId>testng</artifactId>
+            <version>7.4.0</version>
+            <scope>test</scope>
+        </dependency>
+
```
Comment (anushreeSonak) [inline API_Auto/pom.xml]: remove extra space

File: API_Auto/src/test/java/APIAssignments/AllProduct.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import org.apache.logging.log4j.Logger;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+public class AllProduct {
+    private static final Logger log = LogManager.getLogger("AllProduct.class");
+
+    @Test(priority = 1)
+    public void checkStatusCode() {
+        RestAssured.baseURI = "https://automationexercise.com/api/productsList";
```
Comment (anushreeSonak) [inline API_Auto/src/test/java/APIAssignments/AllProduct.java]: Please add URL in Config.properties file and add Config.properties file also

File: API_Auto/src/test/java/APIAssignments/AllProduct.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        Response response = RestAssured.post("https://automationexercise.com/api/productsList")
+                .then().extract().response();
+        Assert.assertEquals(response.getStatusCode(), 200);
+        System.out.println("The response code is : " + response.getStatusCode());
+        log.info("The response code is : " + response.getStatusCode());
+    }
+
+    @Test(priority = 2)
+    public void checkResponse() {
+        Response response = RestAssured.post("https://automationexercise.com/api/productsList")
```
Comment (anushreeSonak) [inline API_Auto/src/test/java/APIAssignments/AllProduct.java]: Please add URL in Config.properties file

File: API_Auto/pom.xml (hunk 1 above)
Comment (waghmaresuyog) [reply API_Auto/pom.xml]: done

File: API_Auto/pom.xml (hunk 2 above)
Comment (waghmaresuyog) [reply API_Auto/pom.xml]: done

File: API_Auto/src/test/java/APIAssignments/AllProduct.java (hunk 3 above)
Comment (waghmaresuyog) [reply API_Auto/src/test/java/APIAssignments/AllProduct.java]: Done

File: API_Auto/src/test/java/APIAssignments/AllProduct.java (hunk 4 above)
Comment (waghmaresuyog) [reply API_Auto/src/test/java/APIAssignments/AllProduct.java]: done

File: API_Auto/pom.xml [hunk 5]
```diff
@@ -3,15 +3,56 @@
...
-    <artifactId>API_Auto</artifactId>
+    <artifactId>APIAssignment</artifactId>
     <version>1.0-SNAPSHOT</version>
-
     <properties>
         <maven.compiler.source>17</maven.compiler.source>
         <maven.compiler.target>17</maven.compiler.target>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     </properties>
-
```
Comment (anushreeSonak) [inline API_Auto/pom.xml]: Remove extra space

File: API_Auto/pom.xml [hunk 6]
```diff
@@ -3,15 +3,56 @@
          xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
          xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
     <modelVersion>4.0.0</modelVersion>
-
     <groupId>org.example</groupId>
-    <artifactId>API_Auto</artifactId>
+    <artifactId>APIAssignment</artifactId>
     <version>1.0-SNAPSHOT</version>
-
```
Comment (anushreeSonak) [inline API_Auto/pom.xml]: Remove extra space

File: API_Auto/src/test/java/APIAssignments/Reader.java [hunk 7]
```diff
@@ -0,0 +1,19 @@
+package APIAssignments;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.Properties;
+
+public class Reader {
+
```
Comment (anushreeSonak) [inline API_Auto/src/test/java/APIAssignments/Reader.java]: Remove extra space

File: API_Auto/pom.xml (hunk 6 above)
Comment (waghmaresuyog) [reply API_Auto/pom.xml]: Done

File: API_Auto/pom.xml (hunk 5 above)
Comment (waghmaresuyog) [reply API_Auto/pom.xml]: done

File: API_Auto/src/test/java/APIAssignments/Reader.java (hunk 7 above)
Comment (waghmaresuyog) [reply API_Auto/src/test/java/APIAssignments/Reader.java]: Done

**Review by anushreeSonak — APPROVED** [review]
Approving PR

**Pull request merged**